packages/hash/frontend/src/blocks/page/Comments/CommentBlockMenu.tsx

```diff
@@ -0,0 +1,62 @@
+import { FunctionComponent, useMemo } from "react";
+import { ListItemIcon, ListItemText } from "@mui/material";
+import { bindMenu, PopupState } from "material-ui-popup-state/hooks";
+import { faLink, faPencil, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { Menu, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { MenuItem } from "../../../shared/ui";
+
+type CommentBlockMenuProps = {
+  popupState: PopupState;
+};
+
+export const CommentBlockMenu: FunctionComponent<CommentBlockMenuProps> = ({
+  popupState,
+}) => {
+  const menuItems = useMemo(
+    () => [
+      {
+        title: "Edit",
+        icon: faPencil,
+        // @todo Commented implement functionality
+        onClick: async () => {},
+      },
+      {
+        title: "Copy Link",
+        icon: faLink,
+        // @todo Commented implement functionality
+        onClick: async () => {},
+      },
+      {
+        title: "Delete Comment",
+        icon: faTrash,
+        // @todo Commented implement functionality
+        onClick: async () => {},
+      },
+    ],
+    [],
+  );
```

Comment:
Have you considered making a `CommentBlockMenuItem` component and doing this declaratively instead of looping over this later?

Reply:
I didn't, is there a benefit to it? I would suggest we don't mind this for now because I already made some changes to this in the next PR to come (on editing comments).

Reply:
Well it's the more react way of doing things… There's no need to create a static array of props to then loop over using those to render – just write a component and use it multiple times